mountpoint-s3/tests/reftests/harness.rs

```diff
@@ -143,189 +280,249 @@ impl Harness {
             offset += num_bytes;
         }
     }
+}
 
-    /// Walk the filesystem tree and check that at each level, contents match the reference
-    async fn compare_contents(&self) {
-        let root = self.reference.root();
-        self.compare_contents_recursive(FUSE_ROOT_INODE, FUSE_ROOT_INODE, root)
-            .await;
+/// Read-only reftests that generate random S3 buckets and check the mapping from S3 keys to file
+/// paths is correct.
+mod read_only {
```

Comment:
GitHub doesn't do a good job rendering this part of the diff, but it's just taking all the existing proptest code and sticking it into a `read_only` module, so we can separate it from the new tests.